### PR TITLE
Remove flake-utils

### PR DESCRIPTION
**Pull request by nrabulinski**
Removes flake utils, simplifies the flake definition, and updates the input to latest `nixpkgs-unstable`

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -2,61 +2,39 @@
...
+    devShells = forEachSystem (pkgs: {
+      default = pkgs.haskellPackages.shellFor {
+        packages = _: [package.${pkgs.system}];
+
+        nativeBuildInputs = [
+          pkgs.cabal-install
+          pkgs.haskellPackages.haskell-language-server
+        ];
+      };
+    });
```
Comment (cdepillabout) [inline flake.nix]: Thanks a lot for fixing this up!

Could you refactor this to make it an overlay (well, a Haskell overlay within a Nixpkgs overlay) on top of Nixpkgs, similar to how the original code is written?

**Comment by Gerg-L** [comment]
Use `?ref` and capitalize `NixOS` in the url :trollface:

File: flake.nix (hunk 1 above)
Comment (Gerg-L) [reply flake.nix]: it doesn't provide an overlay output so that overlay is pointless and requires instantiating a nixpkgs instance

Comment (Gerg-L) [reply flake.nix]: adding
```nix
    overlays.default = final: prev: {
      OverlayedHaskellPackages = final.haskellPackages.override {
        overrides = hfinal: hprev: {
          # This is our local Haskell package.
          example-haskell-package = hfinal.callCabal2nix "example-haskell-nix-flake" ./. {};
        };
      };
    };
```
will give you a usable external overlay

Comment (nrabulinski) [reply flake.nix]: I wanted to do that but override the actual Haskell packages and also provide the binary in the overlay, but that's after I come back from vacation so in 3 days

**Comment by Gerg-L** [comment]
Why instantiation a nixpkgs instance for a `packages` output?
IMO have the overlay as it's own output. But don't consume it for your packages

**Comment by nrabulinski** [comment]
> Why instantiation a nixpkgs instance for a `packages` output? IMO have the overlay as it's own output. But don't consume it for your packages

because my options are either to repeat myself when declaring the packages or a sketchy abstraction

**Comment by cdepillabout** [comment]
Thanks, I appreciate this!

**Pull request merged**